nav2_collision_monitor/params/collision_monitor_params.yaml

```diff
@@ -42,11 +42,45 @@ collision_monitor:
     FootprintApproach:
       type: "polygon"
       action_type: "approach"
-      footprint_topic: "/local_costmap/published_footprint"
+      # footprint_topic: "/local_costmap/published_footprint"
       time_before_collision: 2.0
       simulation_time_step: 0.1
       min_points: 6
-      visualize: False
+      visualize: True
+      polygon_pub_topic: "polygon_approach"
+      polygon_velocity: ["stop", "rotation", "translation_forward", "translation_backward"] # third priority
+      stop:
+        points: [0.25, 0.25, 0.25, -0.25, -0.25, -0.25, -0.25, 0.25]
+        linear_min: 0.0
+        linear_max: 0.01
+        direction_min: -3.14159
+        direction_max: 3.14159
```

Comment:
I don't understand "direction" min / max. If we have the linear velocity min/max and the angular velocity min/max (which should be signed), why is this relevant?

Reply:
As I understood, this to be a sector of the velocities direction whether the required dynamic polygon actions are applied to:
![directions](https://github.com/ros-planning/navigation2/assets/60094858/107dcc7b-a5cf-4182-acdd-694c0f9aad63)

This might be useful, if velocity is not always parallel to OX coordinate in robot base frame, e.g. for omni-drives.

Reply:
I don't understand, velocity is signed, why don't we just specify velocity ranges with signs? That should still establish the cmd_vel range. I don't think we should overoptimize here to get this out. We can always come back if there's a tangible issue for omni robot platforms. 

Though, I don't really see this example being really useful to even omni robot providers. They can still describe the polygon to use if the X velocity is in some range and the Y velocity is in some range, so that would still describe this envelope (e.g. atan2(y,x) gives you the relative angle -- so its redundant information)

Reply:
This might be overoptimiation, if we could use the velocity corridors in velocity сartesian coordinate system. However, how could we set this space if we want to reflect some 45 degrees direction (e.g. for omni-drives) with velocity from 0 ... to max? From other side I am no sure, that this has a real use-case applications - from initial talks, we we used to say about forward/backward moving directions and rotation in a place. Plus your example when we could neglect one of the velocity components (that could not be resolved in polar coordinate system).

Thinking twice about it: yes, this seems to be unnecessary complication.

Reply:
I'd be curious what @kaichie thinks. 

OK, I can get on board with this. I think we should have some param for `holonomic` and if `true` we consider `direction`, if not that part of of the `isInRange` is skipped. While that's not a really formal abstraction of the different motion model types, I think that's sufficient. 

I think including a good piece of documentation in the readme + a version of that diagram you provided will be important to describe what that means in detail for omni robots

Reply:
For holonomic platform, I personally believe that it will be more user friendly to define the `linear` magnitude and `direction` rather than two separate ranges in `x` and `y`. It will be valuable to gather insights from the holonomic platform community to determine the more advantageous approach.

Sounds good to add the flag to simplify the process of configuring polygons for non-holonomic robot types.

As for the `direction_min` and `direction_max` configuration, my current concept and implementation assume that the polygon's covered direction always spans from `direction_min` to `direction_max`. With this, we can specify the angle within any range. If we are defining the range from `0 to min` and `0 to max`, then we are not able to specify ranges like the backward direction. For example:

![velocityranges](https://github.com/ros-planning/navigation2/assets/37255935/621778d4-f6cf-4ed2-88be-a1c7203b4a7b)

Reply:
I am OK with the idea of using `holonomic` parameter which specifies which restrictions are we operating with: polar-based `direction_min/direction_max` + `linear_min/linear_max`; or cartesian-based `vel_x_min/vel_x_max` + `vel_y_min/vel_y_max` for differential and Ackermann/legged drives.
This is some kind of controller-like approach.

From another hand, to not make Collision Monitor to be controller-independent, as it originally conceived, I'd like to propose to make `direction_min`/`direction_max` to be an optional parameters. If one of them was specified, we will use their values. If none of them were specified - check velocity component restrictions only (and thus, do not do `std::atan()` cost-ineffective operations).
Moreover, if we will use polar-based and cartesian-based restrictions at the same time, it could allow us to be even more flexible in making robot velocity application rules (say, forward movement in 15 degree sector, but x-component of speed no less than some minimal value with total ignoring of y-value of velocity).

@kaichie, and yes, the clarification of directions and its signs (clock/counter clock wise rotations) and the pictures from above would be extremely helpful to include it into final documentation.

Reply:
> I personally believe that it will be more user friendly to define the linear magnitude and direction rather than two separate ranges in x and y.

I would agree conceptually in reducing the number of parameters down to a minimum. It would though make configuring a bit more painful, but I don't see a way around that. 

> Sounds good to add the flag to simplify the process of configuring polygons for non-holonomic robot types.

:+1: So only check on the direction bits when `holonomic = true`, else only consider `X` and `theta` vel.

> As for the direction_min and direction_max configuration, my current concept and implementation assume that the polygon's covered direction always spans from direction_min to direction_max. With this, we can specify the angle within any range. If we are defining the range from 0 to min and 0 to max, then we are not able to specify ranges like the backward direction

I don't understand. If we call the direction from -PI to PI with `0` being forward, then reverse would be from -PI/2 to PI/2. This is why I asked in another part of the code review to review how you handle the ranges of direction when checking if the polygon should be applied for wrap around. Just checking `> min` and `< max` doesn't work with angles. 

>  This is some kind of controller-like approach.

I don't think this is a controller approach, this is just an approach that is cognizant of robot kinematic constraints - which this would obviously need to have since different robots move differently and thus would represent their motion abilities to define polygons for separately. If we had to change anything more then `direction` checking, I would ask for an abstraction of the motion models in this node. But while its simple and only 1 change, I think its fine as a global param as to which behavior to use.

Reply:
Just to summarise the proposed ideas here, I think so far these are the ideas that we have agreed on:
- Add `holonomic` flag
- If `holonomic` is `false`: consider velocity `x` within `vel_x_min`/`vel_x_max` and `theta_min/theta_max`

The ideas still ongoing for discussion where `holonomic` is `true` are as follows:
**Option 1:**  consider velocity within `linear_min/linear_max`, `direction_min/direction_max` and `theta_min/theta_max`  
**Option 2:** consider velocity within `vel_x_min/vel_x_max/vel_y_min/vel_y_max` and `theta_min/theta_max`.
**Option 3:** consider velocity within `linear_min/linear_max` and `direction_min/direction_max`(if provided as an optional parameter). Otherwise, consider `vel_x_min/vel_x_max/vel_y_min/vel_y_max`. +  `theta_min/theta_max`

I think **Option 3** would cover most of the concern we have discussed, offering the flexibility to define either within a linear+direction range or using a Cartesian-based range approach. 

For the direction range , lets discuss on the other part of the code review. I will include the table for all the possible range options(probably tomorrow).

Reply:
I think we need to do Option 1. As proposed Option 3 is Option 1 but without Theta, but keep in mind translational velocity in some component direction doesn't preclude rotations at the same time, so we still need to consider a rotational component. 

Option 3 seems to cover 1 and 2 as you mention, but it provides redundent methods of embedding the same information which should be avoided unless there are reasons to have both (e.g. some things you can't model in one you need the other for)

Reply:
I will proceed with Option 1 to make the relevant changes. Much appreciated for the discussion and clarification @AlexeyMerzlyakov @SteveMacenski! :pray: I will include the related diagrams in the documentation as discussed.